statsmodels/graphics/gofplots.py

```diff
@@ -599,41 +620,60 @@ def qqline(ax, line, x=None, y=None, dist=None, fmt='r-'):
     dist : scipy.stats.distribution
         A scipy.stats distribution, needed if line is 'q'.
 
+    Returns
+    -------
+    Matplotlib line artist.
+
     Notes
     -----
     There is no return value. The line is plotted on the given `ax`.
+
     """
+
+    valid_lines = ['45', 'q', 'r', 's']
+    if line not in valid_lines:
+        raise ValueError('`line` must be one of {0}'.format(valid_lines))
+
     if line == '45':
         end_pts = lzip(ax.get_xlim(), ax.get_ylim())
         end_pts[0] = min(end_pts[0])
         end_pts[1] = max(end_pts[1])
-        ax.plot(end_pts, end_pts, fmt)
+        lineartist, = ax.plot(end_pts, end_pts, fmt, **lineoptions)
         ax.set_xlim(end_pts)
         ax.set_ylim(end_pts)
-        return # does this have any side effects?
-    if x is None and y is None:
-        raise ValueError("If line is not 45, x and y cannot be None.")
-    elif line == 'r':
-        # could use ax.lines[0].get_xdata(), get_ydata(),
-        # but don't know axes are 'clean'
-        y = OLS(y, add_constant(x)).fit().fittedvalues
-        ax.plot(x,y,fmt)
-    elif line == 's':
-        m,b = y.std(), y.mean()
-        ref_line = x*m + b
-        ax.plot(x, ref_line, fmt)
-    elif line == 'q':
-        _check_for_ppf(dist)
-        q25 = stats.scoreatpercentile(y, 25)
-        q75 = stats.scoreatpercentile(y, 75)
-        theoretical_quartiles = dist.ppf([0.25, 0.75])
-        m = (q75 - q25) / np.diff(theoretical_quartiles)
-        b = q25 - m*theoretical_quartiles[0]
-        ax.plot(x, m*x + b, fmt)
+
+    else:
+        if x is None or y is None:
+            raise ValueError("If line is not 45, x and y cannot be None.")
+        else:
+            x = np.array(x)
+            y = np.array(y)
+
+        if line == 'r':
+            # could use ax.lines[0].get_xdata(), get_ydata(),
+            # but don't know axes are 'clean'
+            y = OLS(y, add_constant(x)).fit().fittedvalues
+            lineartist, = ax.plot(x, y, fmt, **lineoptions)
+
+        elif line == 's':
+            m, b = np.std(y), np.mean(y)
+            ref_line = x*m + b
+            lineartist, = ax.plot(x, ref_line, fmt, **lineoptions)
+
+        elif line == 'q':
+            _check_dist(dist)
+            q25 = stats.scoreatpercentile(y, 25)
+            q75 = stats.scoreatpercentile(y, 75)
+            theoretical_quartiles = dist.ppf([0.25, 0.75])
+            m = (q75 - q25) / np.diff(theoretical_quartiles)
+            b = q25 - m*theoretical_quartiles[0]
+            lineartist, = ax.plot(x, m*x + b, fmt, **lineoptions)
+
+    return lineartist
 
 
 #about 10x faster than plotting_position in sandbox and mstats
-def plotting_pos(nobs, a):
+def plotting_pos(nobs, a=0, b=0):
```

Comment:
better default `b=None` which uses `b=a`
I think that will be backwards compatible and easier to use

plotting position need review everywhere for addition of b, docstrings ? (maybe resolved, I was browsing individual commits)